### PR TITLE
[CL-269] extension router transitions

### PR DESCRIPTION
**Pull request by willmartian**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/CL-269

## 📔 Objective

Update the browser extension router animations to 1.) look different and 2.) use an _elevation_ concept instead of defining every page navigation possibility in the animation config file. 

Routes in the extension must now declare an elevation in their data:

```ts
/**
 * Determines the router transition behavior.
 * Changing between elevations will animate from the right.
 * Navigating between two pages of the same elevation will not animate.
 */
export type RouteElevation = 0 | 1 | 2 | 3 | 4;

// app-routing.module.ts
{
    path: "set-password",
    component: SetPasswordComponent,
    data: { elevation: 1 } satisfies RouteDataProperties,
  },
```

(Also, the DS team is taking code ownership of this unless anyone wants to fight me for it 🥊 )

## 📸 Screenshots

https://github.com/user-attachments/assets/d5e3d608-d49e-4b3e-99aa-10c123e54b0f

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: apps/browser/src/popup/app-routing.module.ts
Comment (willmartian) [inline apps/browser/src/popup/app-routing.module.ts:1]: Regexp find & replace for the win 😎

**Comment by willmartian** [comment]
@vleague2 I still need to tweak the actual animation that is being displayed, but what do you think of this elevation abstraction?

File: apps/browser/src/popup/app.component.ts [hunk 1]
```diff
@@ -220,23 +220,12 @@ export class AppComponent implements OnInit, OnDestroy {
     this.destroy$.complete();
   }
 
-  getState(outlet: RouterOutlet) {
+  getRouteElevation(outlet: RouterOutlet) {
     if (!this.routerAnimations) {
       return;
-    } else if (outlet.activatedRouteData.state === "ciphers") {
```
Comment (vleague2) [inline apps/browser/src/popup/app.component.ts:226]: Is all this cipher stuff covered in the new elevation flows?

**Review by vleague2 — COMMENTED** [review]
I like the approach!! Looks much simpler and consistent

File: apps/browser/src/popup/app.component.ts (hunk 1 above)
Comment (willmartian) [reply apps/browser/src/popup/app.component.ts:226]: Yes. It isn't necessary anymore.

File: apps/browser/src/popup/app-routing.animations.ts [hunk 2]
```diff
@@ -30,214 +50,53 @@ export function queryTranslate(
...
   );
 }
 
-export function queryTranslateX(direction: string, from: number, to: number, zIndex = 1000) {
-  return queryTranslate(direction, "X", from, to, zIndex);
-}
-
-export function queryTranslateY(direction: string, from: number, to: number, zIndex = 1000) {
-  return queryTranslate(direction, "Y", from, to, zIndex);
-}
```
Comment (willmartian) [inline apps/browser/src/popup/app-routing.animations.ts:45]: These bound functions confused me more than helped.

**Review by vleague2 — APPROVED** [review]
Love it

**Review by danielleflinn — CHANGES_REQUESTED** [review]
✅ Animations slide in from left, on top of previous page for majority of pages
✅ No animation when navigating between the View and Edit item pages 
✅ Account switcher slides in from top 
⛔️ When viewing an item in the Trash, the "Trash" view slides out to reveal the View item page; then slides back over the View item when the user selects "back". The View item page should slide in, on top of the "Trash" view, then slide out when the user navigates back.

**Comment by willmartian** [comment]
> ⛔️ When viewing an item in the Trash, the "Trash" view slides out to reveal the View item page; then slides back over the View item when the user selects "back". The View item page should slide in, on top of the "Trash" view, then slide out when the user navigates back.

Fixed with [aee101c](https://github.com/bitwarden/clients/pull/11989/commits/aee101c051dc6bbc37a6f37e49e32c0fa0c77e60) @danielleflinn

**Review by danielleflinn — CHANGES_REQUESTED** [review]
@willmartian thanks for fixing the Trash view bug. I got some last minute feedback, for now we want to update the Account switcher transition to also be slide in from right to remain consistent with the other animations. Once we have that change we are good to go to merge. Thanks!

**Review by vleague2 — APPROVED** [review]

**Review by danielleflinn — APPROVED** [review]
This looks great! Thanks for all your efforts with this on @willmartian 🎉

**Pull request merged**